front-end/src/components/analysis-page/IntentMenu.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import store from "../../store.js";
+import { setDisplayingQuestion } from '../../store/analyzeTranscriptSlice.js';
+
+class IntentMenu extends Component {
+    
+    constructor(props) {
+        super(props);
+        this.ref = React.createRef();
```

Comment:
I would avoid using a ref instead of the state. Ref's do not test quite as nicely from my experience (although this might be different with react testing library instead of enzyme)

Reply:
I will see what I can do. The tutorial for checking clicks outside of an element used Ref's but I didn't think too hard on whether they were actually needed.

Reply:
I've checked 4 different tutorials which all use the ref. Is there a way to do it thru state?

Also to respond to the related comment below - I _think_ ref is used here to tag the entire intent menu element. Then, whenever the user clicks somewhere, the click handler checks if the element that the user clicked on is contained by the menu (which is identified through ref). Is there a non-ref way to do this?